### ISSUE TITLE
Buggy ruleset behavior?

### ISSUE THREAD

**Issue by cspotcode** [balderdashy/anchor#68]
I'm trying to understand what anchor is doing and I think I've found a big bug.

https://github.com/balderdashy/anchor/blob/master/lib/match/matchType.js#L149
```
return errors.concat(deepMatchType.call(self, data[key], ruleset[key], depth + 1, key, customMessage));
```

As I understand this code, `data` is the Javascript value being validated and `ruleset` is a dictionary of validation rules that must be run against the data.  For example:
```
data = 123;
ruleset = {required: true, type: 'number', greaterThan: 10};
```

If I'm correct, shouldn't the linked code be:
```
return errors.concat(deepMatchType.call(self, data, ruleset[key], depth + 1, keyName, customMessage));
```

We should not be indexing into data using the name of a rule; it doesn't make sense to do `(123).required`.

For that matter, should we be calling `deepMatchType`?  What about `matchRule`?

**Comment by sailsbot** [comment]
Thanks for posting, @cspotcode.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
